src/modelio.cs

```diff
@@ -1571,12 +1667,26 @@ interface MDLTexture : MDLNamed
 		[Export ("writeToURL:")]
 		bool WriteToUrl (NSUrl url);
 
+		[TV (11,0), Mac (10,13, onlyOn64:true), iOS (11,0)]
+		[Export ("writeToURL:level:")]
+		bool WriteToUrl (NSUrl url, nuint level);
+
 		[Export ("writeToURL:type:")]
 		bool WriteToUrl (NSUrl url, string type);
 
-		[NullAllowed, Export ("imageFromTexture")]
+		[TV (11,0), Mac (10,13, onlyOn64:true), iOS (11,0)]
+		[Export ("writeToURL:type:level:")]
+		bool WriteToUrl (NSUrl nsurl, string type, nuint level);
+
+		[Export ("imageFromTexture")]
+		[return: NullAllowed]
 		CGImage GetImageFromTexture ();
 
+		[TV (11,0), Mac (10,13, onlyOn64:true), iOS (11,0)]
+		[Export ("imageFromTextureAtLevel:")]
+		[return: NullAllowed]
+		CGImage GetImageFromTexture (nuint atLevel);
```

Comment:
It's kinda weird to name it `atLevel`. `level` should be good enough.

You then do `atLevel = 3`...